src/calendar/calendar-year.tsx

```diff
@@ -0,0 +1,76 @@
+import { useEffect, useState } from "react";
+import dayjs, { Dayjs } from "dayjs";
+import { CalendarYearProps, VariantYear } from "./types";
+import { YearCell, YearPickerContainer } from "./calendar-year.style";
+import { CalendarHelper } from "../util/calendar-helper";
+
+export const CalendarYear = ({
+    calendarDate,
+    showView,
+    selectedStartDate,
+    onSelect,
+    onDecadeChange,
+}: CalendarYearProps) => {
+    const [yearDate, setYearDate] = useState<Dayjs[]>([]);
+
+    useEffect(() => {
+        if (showView === "Year") {
+            generateDecadeOfYears();
+        }
+    }, [showView, calendarDate]);
+
+    const generateYearStatus = (date: Dayjs) => {
+        const grayYearIndex = [0, 11];
+
+        const isDecadeYear = grayYearIndex.includes(yearDate.indexOf(date));
```

Comment:
can help to rename these variables to reflect their use
- `grayYearIndex` to `otherDecadeIndexes`
- `isDecadeYear` to `isOtherDecade`